src/Engine/GameEvents/CommanderAbilityEvent.java

```diff
@@ -33,7 +36,14 @@ public GameEventQueue sendToListener(GameEventListener listener)
   @Override
   public void performEvent(MapMaster gameMap)
   {
-    myAbility.activate(gameMap);
+    if( myCommander.getAbilityPower() < myAbility.getCost() )
+    {
+      System.out.println("WARNING!: Performing ability with insufficient ability power!");
+    }
+    myCommander.activateAbility(myAbility, gameMap);
```

Comment:
`CommanderAbility.activate()` also calls `Commander.activateAbility()` and does the cost adjustment. This function in general is doing a lot more than it did before, and now it's duplicative.

Should we revert this event and the corresponding action to how they were before? This appears to be residual change from when you pulled Commanders out of `CommanderAbility`.